### PR TITLE
[ENH] Added n_clusters attribute to TSFreshClusterer

### LINKED ISSUE #2199: [ENH] Cant set number of clusters with the default TSFreshClusterer
### Describe the feature or idea you want to propose

TSFreshClusterer takes an estimator argument which defaults to KMeans in fit. You cannot currently set the number of clusters directly. 

### Describe your proposed solution

I think you should be able to set it in the constructor, then use it here in fit 

```python
        self._estimator = _clone_estimator(
            (KMeans() if self.estimator is None else self.estimator),
            self.random_state,
        )
```

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aryanpola**
#### Reference Issues/PRs

Fixes #2199 

#### What does this implement/fix? Explain your changes.

Added `n_clusters` attribute to `TSFreshClusterer` constructor and passed it onto estimator in fit.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

File: aeon/clustering/feature_based/_tsfresh.py [hunk 1]
```diff
@@ -43,6 +45,8 @@ class TSFreshClusterer(BaseClusterer):
         If `RandomState` instance, random_state is the random number generator;
         If `None`, the random number generator is the `RandomState` instance used
         by `np.random`.
+    n_clusters : int, default=20
```
Comment (TonyBagnall) [inline aeon/clustering/feature_based/_tsfresh.py:48]: probably best to use the scikit default of n_clusters = 8

File: aeon/clustering/feature_based/_tsfresh.py [hunk 2]
```diff
@@ -90,13 +95,14 @@ def __init__(
         self.n_jobs = n_jobs
         self.chunksize = chunksize
         self.random_state = random_state
+        self.n_clusters = n_clusters if n_clusters is not None else 20
```
Comment (TonyBagnall) [inline aeon/clustering/feature_based/_tsfresh.py:98]: you should just have
self.n_clusters = n_clusters

File: aeon/clustering/feature_based/_tsfresh.py [hunk 3]
```diff
@@ -121,10 +127,19 @@ def _fit(self, X, y=None):
             n_jobs=self._n_jobs,
             chunksize=self.chunksize,
         )
-        self._estimator = _clone_estimator(
-            (KMeans() if self.estimator is None else self.estimator),
-            self.random_state,
-        )
+
```
Comment (TonyBagnall) [inline aeon/clustering/feature_based/_tsfresh.py:130]: no need to test n_clusters is not None, as it defaults to 8, you can just pass it directly to KMeans without the test

File: aeon/clustering/feature_based/_tsfresh.py [hunk 4]
```diff
@@ -121,10 +127,19 @@ def _fit(self, X, y=None):
...
+        if self.estimator is None:
+            if self.n_clusters is not None:
+                self._estimator = _clone_estimator(
+                    KMeans(n_clusters=self.n_clusters), self.random_state
+                )
+            else:
+                self._estimator = _clone_estimator(KMeans(), self.random_state)
+        else:
+            if hasattr(self.estimator, "n_clusters") and self.n_clusters is not None:
```
Comment (TonyBagnall) [inline aeon/clustering/feature_based/_tsfresh.py:139]: no need to test not None

File: aeon/clustering/feature_based/_tsfresh.py [hunk 5]
```diff
@@ -90,13 +95,14 @@ def __init__(
...
         self.random_state = random_state
+        self.n_clusters = n_clusters if n_clusters is not None else 20
 
         self._transformer = None
         self._estimator = None
 
         super().__init__()
 
-    def _fit(self, X, y=None):
+    def _fit(self, X: np.ndarray, y: Optional[np.ndarray] = None) -> "TSFreshClusterer":
```
Comment (MatthewMiddlehurst) [inline aeon/clustering/feature_based/_tsfresh.py:105]: this is not a good type hint for the output, the function does not return a string. Unless this is some method I'm not sure about for hinting its returning an object (self). Bad hints are worse then no hints.

**Comment by aryanpola** [comment]
@TonyBagnall @MatthewMiddlehurst I've set the `n_clusters` as 8, but when `None` is passed on to n_clusters, it's not taking as 8.
